boards/renesas/rzg3s_smarc/doc/index.rst

```diff
@@ -0,0 +1,250 @@
+
+  .. code-block:: bash
+    :emphasize-lines: 6
+
+      require trusted-firmware-a.inc
+      COMPATIBLE_MACHINE_rzg3s = "(rzg3s-dev|smarc-rzg3s)"
+      PLATFORM_rzg3s-dev = "g3s"
+      EXTRA_FLAGS_rzg3s-dev = "BOARD=dev14_1_lpddr PLAT_SYSTEM_SUSPEND=vbat"
+      PLATFORM_smarc-rzg3s = "g3s"
+      EXTRA_FLAGS_smarc-rzg3s = "BOARD=smarc PLAT_SYSTEM_SUSPEND=vbat PLAT_M33_BOOT_SUPPORT=1"
+
+  3. Start the build:
+
+  .. code-block:: bash
+
+      MACHINE=smarc-rzg3s bitbake core-image-minimal
+
+  The below necessary artifacts will be located in the build/tmp/deploy/images
+
+  +---------------+-----------------------------+
+  | Artifacts     | File name                   |
+  +===============+=============================+
+  | Boot loader   | bl2_bp_spi-smarc-rzg3s.srec |
+  |               |                             |
+  |               | fip-smarc-rzg3s.srec        |
+  +---------------+-----------------------------+
+  | Flash Writer  | FlashWriter-smarc-rzg3s.mot |
+  +---------------+-----------------------------+
+
+  4. Follow ''4.2 Startup Procedure'' of `SMARC EVK of RZ/G3S Linux Start-up Guide`_ for power supply and board setting
+     at SCIF download (SW_MODE[1:4] = OFF, ON, OFF, ON) and Cortex-A55 cold boot (SW_CONFIG[1:6] = OFF, OFF, ON, OFF, OFF, OFF)
+
+  5. Follow ''4.3 Download Flash Writer to RAM'' of `SMARC EVK of RZ/G3S Linux Start-up Guide`_ to download Flash Writer to RAM
+
+  6. Follow ''4.4 Write the Bootloader'' of `SMARC EVK of RZ/G3S Linux Start-up Guide`_ to write the boot loader
+     to the target board by using Flash Writer.
+
+Applications for the ``rzg3s_smarc`` board can be built in the usual way as
+documented in :ref:`build_an_application`.
+
+Console
+=======
+
+The UART port for Cortex-M33 System Core can be accessed by connecting `Pmod USBUART <https://store.digilentinc.com/pmod-usbuart-usb-to-uart-interface/>`_
+to the upper side of ``PMOD1_3A``.
+
+Debugging
+=========
+
+It is possible to load and execute a Zephyr application binary on
+this board on the Cortex-M33 System Core from
+the internal SRAM, using ``JLink`` debugger (:ref:`jlink-debug-host-tools`).
+
+.. note::
+
+    Currently it's required Renesas BL2 TF-A to be started on Cortex-A55 System Core
+    before starting Zephyr. As BL2 TF-A configures clocks and Cortex-M33 System Core before starting it.
+
+Here is an example for building and debugging with the :zephyr:code-sample:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: rzg3s_smarc/r9a08g045s33gbg/cm33
+   :goals: build debug
+
+Flashing
+========
+
+Zephyr application can be flashed to QSPI storage and then loaded by
+Renesas BL2 TF-A running on Cortex-A55 System Core and staring binary at Cortex-M33 System Core.
+
+The Zephyr application binary has to be converted to Motorolla S-record `SREC`_ format
+which is generated automatically in Zephyr application build directory (zephyr.s19).
```

Comment:
```suggestion
which is generated automatically in Zephyr application build directory.
```
the name is determined by a Kconfig

Reply:
I removed the fixed name and mentioned the file extension s19.